lib/extractor/docker-archive/index.ts

```diff
@@ -15,7 +16,13 @@ export function getImageIdFromManifest(
   manifest: DockerArchiveManifest,
 ): string {
   try {
-    return manifest.Config.split(".")[0];
+    const imageId = manifest.Config.split(".")[0];
+    const hashAlgorithmPrefix = `${HashAlgorithm.Sha256}:`;
+    if (imageId.includes(":")) {
```

Comment:
isn't it always supposed to include the algorithm according to the spec?

Reply:
The behavior today is that we take the imageId from the manifest.json Config attribute's value, which is the imageSha without the prefix, with a '.json' suffix. i.e. today there is no prefix.
As [mentioned](https://snyk.slack.com/archives/C01EPSMGZML/p1618420140094600?thread_ts=1618396754.091000&cid=C01EPSMGZML) by @agatakrajewska, the documentation of docker save is not the best, and so I added this check to make sure we don't accidentally end up with having the prefix twice, although I don't have a real example for this.

Reply:
I'm missing something, the spec for the manifest's config is described here: https://github.com/opencontainers/image-spec/blob/master/manifest.md
the ```config``` component is described as a ``` descriptor``` which is defined here: https://github.com/opencontainers/image-spec/blob/master/descriptor.md
its properties include a digest which is defined here: https://github.com/opencontainers/image-spec/blob/master/descriptor.md#digests
(sorry for the long link chain 😂)
but it says the digest must adhere to this grammar:
```
digest                ::= algorithm ":" encoded
algorithm             ::= algorithm-component (algorithm-separator algorithm-component)*
algorithm-component   ::= [a-z0-9]+
algorithm-separator   ::= [+._-]
encoded               ::= [a-zA-Z0-9=_-]+
```
so must always include the algorithm.

where is the discrepancy?

Reply:
First I'd mention that we are talking here about docker-archive and not oci-archive. In OCI archive we indeed get the image ID from manifest.config.digest, which includes the algorithm prefix.
As @agatakrajewska mentioned [here](https://snyk.slack.com/archives/C01EPSMGZML/p1618420140094600?thread_ts=1618396754.091000&cid=C01EPSMGZML):
'docker save does not save into manifest v2, schema2, they use something custom based on v1'.
Therefore, in the docker archive, we get the imageId from the manifest.json 'Config' attribute value, which is actually the path of the config.json file in the archive. The name of this file is the imageId (without the algo prefix). 

From what I understand from the links I sent above, the algorithm itself is sha256, although this information is not mentioned in the docker archive itself. 

I think it would be best to have @agatakrajewska 's input here 🙏

Reply:
okay, makes sense :) feel free to resolve

Reply:
Sharon, you nailed it! I think the only uncertainty (is that a word?) is are we sure that it's always sha256 algorythm? But based on the docs you found I think it's pretty safe to make that assumption.